api/tests/app/test_app.py

```diff
@@ -0,0 +1,150 @@
+        ({"occupancy_alerts_min_interval": "Here_must_be_an_integer_variable"}, "integer"),
+        ({"max_thread_restarts": False}, "integer"),
+        ({"max_thread_restarts": "Here_must_be_an_integer_variable"}, "integer"),
+        ({"global_reporting_emails": 40}, "string"),
+        ({"global_reporting_emails": False}, "string"),
+        ({"global_report_time": 40}, "string"),
+        ({"global_report_time": False}, "string"),
+        ({"daily_global_report": "Here_must_be_a_bool_variable"}, "bool"),
+        ({"daily_global_report": 40}, "bool"),
+        ({"weekly_global_report": "Here_must_be_a_bool_variable"}, "bool"),
+        ({"weekly_global_report": 40}, "bool"),
+        ({"heatmap_resolution": 40}, "string"),
+        ({"heatmap_resolution": False}, "string"),
+    ])
+    def test_try_change_app_config_wrong_type_variable_all(self, config_rollback, key_value_dict,
+                                                           correct_type):
+        """
+        It is important to mention, the fact that if pydantic expects a string, it will try to
+        cast what was sent, so if we send a bool or an integer to a string field, pydantic will work:
+          Example:
+              a = bool or integer
+              str(a)
+          Pydantic will accept it and PUT request will work.
+
+        Idem when we send a bool, and an integer is expected.
+            int(False) = 0
+            int(True) = 1
+        """
+        client, config_sample_path = config_rollback
+
+        body = create_app_config(key_value_dict)
+
+        response = client.put('/app', json=body)
+
+        if correct_type == "string":
+            assert response.status_code == 200
+            expected_response = get_app_from_config_file(config_sample_path)
+            key = list(key_value_dict.keys())[0]
+            value = list(key_value_dict.values())[0]
+            expected_response[key] = str(value)
+
+            assert response.json() == expected_response
+        elif correct_type == "integer":
+            for key, value in key_value_dict.items():
+                if isinstance(key_value_dict[key], str):
+                    assert response.status_code == 400
+                    assert response.json()['detail'][0]['type'] == 'type_error.' + correct_type
+                else:
+                    assert response.status_code == 200
+                    expected_response = get_app_from_config_file(config_sample_path)
+                    assert response.json() == expected_response
+        elif correct_type == "bool":
+            assert response.status_code == 400
+            assert response.json()['detail'][0]['type'] == 'type_error.' + "bool"
+        else:
+            assert response.status_code == 400
+
+    # pytest -v api/tests/app/test_app.py::TestClassUpdateAppConfig::test_try_change_app_config_non_existence_key
+    def test_try_change_app_config_non_existence_key(self, config_rollback):
```

Comment:
We could add a docstring or comment that extra keys are ignored

Reply:
Changed!